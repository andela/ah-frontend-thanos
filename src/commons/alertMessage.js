const alertMessage = {
  title: 'Problem encountered',
  text: '',
  type: 'error',
  confirmButtonText: 'OK',
};

export default alertMessage;
