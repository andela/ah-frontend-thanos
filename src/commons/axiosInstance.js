import axios from 'axios';

const baseURL = 'https://ah-backend-thanos-stagin-pr-48.herokuapp.com/';
const timeout = 60000;
const token = localStorage.getItem('token');
const headers = token
  ? {
    'Content-Type': 'application/json',
    Authorization: `Token ${token}`,
  } : {
    'Content-Type': 'application/json',
  };
const axiosInstance = axios.create({
  baseURL,
  timeout,
  headers,
});
export default axiosInstance;
