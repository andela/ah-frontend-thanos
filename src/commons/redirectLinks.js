// redirect links at the bottom of signup and login pages
import React from 'react';

const generateRedirectLinks = (message, link, linkName) => (
  <div className="small mt-3">
    <span>{message}</span>
    <a href={`/${link}`}>{linkName}</a>
  </div>
);

export default generateRedirectLinks;
