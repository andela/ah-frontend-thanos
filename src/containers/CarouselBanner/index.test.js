import React from 'react';
import { shallow } from 'enzyme';
import { CarouselBanner } from '.';
import { sampleListOfArticles } from '../../commons/initialStates';

describe('<CarouselBanner /> container', () => {
  let wrapper;
  let wrapper2;
  let articles;

  beforeEach(() => {
    articles = { results: sampleListOfArticles };
    wrapper = shallow(<CarouselBanner articles={articles} />);
    wrapper2 = shallow(<CarouselBanner articles={{ articles: {} }} />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper2).toMatchSnapshot();
  });

  test('Test for a class in a wrapper', () => {
    expect(wrapper.find('.carousel-item')).toHaveLength(3);
  });

  test('Test for a class in a wrapper', () => {
    expect(wrapper.find('.active')).toHaveLength(2);
  });
});
