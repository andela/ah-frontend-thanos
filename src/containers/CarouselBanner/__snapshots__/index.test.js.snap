// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselBanner /> container should render 1`] = `
<div
  className="carousel slide"
  data-ride="carousel"
  id="carouselExampleIndicators"
>
  <ol
    className="carousel-indicators"
  >
    <li
      className="active"
      data-slide-to="0"
      data-target="#carouselExampleIndicators"
    />
    <li
      data-slide-to="1"
      data-target="#carouselExampleIndicators"
    />
    <li
      data-slide-to="2"
      data-target="#carouselExampleIndicators"
    />
  </ol>
  <div
    className="carousel-inner"
  >
    <div>
      <div
        className="carousel-item carousel-image-item active"
        key="1"
      >
        <img
          alt="slide"
          className="d-block w-100"
          src="http://test.png"
        />
        <div
          className="carousel-caption d-none d-md-block carousel-article-desc"
        >
          <h5>
            <a
              href="article/1"
            >
              title sample
            </a>
          </h5>
          <p>
            desc
          </p>
        </div>
      </div>
      <div
        className="carousel-item carousel-image-item "
        key="2"
      >
        <img
          alt="slide"
          className="d-block w-100"
          src="http://test.png"
        />
        <div
          className="carousel-caption d-none d-md-block carousel-article-desc"
        >
          <h5>
            <a
              href="article/2"
            >
              title sample
            </a>
          </h5>
          <p>
            desc
          </p>
        </div>
      </div>
      <div
        className="carousel-item carousel-image-item "
        key="3"
      >
        <img
          alt="slide"
          className="d-block w-100"
          src="http://test.png"
        />
        <div
          className="carousel-caption d-none d-md-block carousel-article-desc"
        >
          <h5>
            <a
              href="article/3"
            >
              title sample
            </a>
          </h5>
          <p>
            desc
          </p>
        </div>
      </div>
    </div>
  </div>
  <a
    className="carousel-control-prev"
    data-slide="prev"
    href="#carouselExampleIndicators"
    role="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-prev-icon"
    />
    <span
      className="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    className="carousel-control-next"
    data-slide="next"
    href="#carouselExampleIndicators"
    role="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-next-icon"
    />
    <span
      className="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;

exports[`<CarouselBanner /> container should render 2`] = `
<div
  className="carousel slide"
  data-ride="carousel"
  id="carouselExampleIndicators"
>
  <ol
    className="carousel-indicators"
  >
    <li
      className="active"
      data-slide-to="0"
      data-target="#carouselExampleIndicators"
    />
    <li
      data-slide-to="1"
      data-target="#carouselExampleIndicators"
    />
    <li
      data-slide-to="2"
      data-target="#carouselExampleIndicators"
    />
  </ol>
  <div
    className="carousel-inner"
  >
    <span />
  </div>
  <a
    className="carousel-control-prev"
    data-slide="prev"
    href="#carouselExampleIndicators"
    role="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-prev-icon"
    />
    <span
      className="sr-only"
    >
      Previous
    </span>
  </a>
  <a
    className="carousel-control-next"
    data-slide="next"
    href="#carouselExampleIndicators"
    role="button"
  >
    <span
      aria-hidden="true"
      className="carousel-control-next-icon"
    />
    <span
      className="sr-only"
    >
      Next
    </span>
  </a>
</div>
`;
